test/integration/shadow-dom.js

```diff
@@ -0,0 +1,52 @@
+var runWebdriver = require('../run-webdriver'),
+	assert = require('chai').assert,
+	host = 'localhost',
+	AxeBuilder = require('../../lib');
+
+if (process.env.REMOTE_TESTSERVER_HOST) {
+	host = process.env.REMOTE_TESTSERVER_HOST;
+}
+
+var shadowSupported;
+
+describe.only('shadow-dom.html', function () {
+	this.timeout(10000);
+
+	var driver;
+	before(function (done) {
+		driver = runWebdriver();
+		driver.manage().timeouts().setScriptTimeout(500);
+		driver
+			.get('http://' + host + ':9876/test/fixtures/shadow-dom.html')
+			.then(function () {
+				driver.executeAsyncScript(function(callback) {
+          var script = document.createElement('script');
+          script.innerHTML = 'var shadowSupport = document.body && typeof document.body.attachShadow === \'function\';';
+          document.documentElement.appendChild(script);
+        	shadowSupported = shadowSupport;
```

Comment:
@WilcoFiers any ideas on how I can get a variable back out of `executeAsyncScript`? The shadowSupport check is working, but I can't store it in a variable in the outer scope.